test-case/check-signal-stop-start.sh

```diff
@@ -0,0 +1,99 @@
+##    simulate keyboard inputs ctrl+z and fg.
+##    kill SIGSTOP $pid will stop the thread
+##    kill SIGCONT $pid will start the thread
+## Case step:
+##    1. aplay/arecord on PCM
+##    2. use send SIGSTOP/SIGCONT to stop/start
+## Expect result:
+##    no error happen for aplay/arecord
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['t']='tplg'     OPT_DESC_lst['t']='tplg file, default value is env TPLG: $TPLG'
+OPT_PARM_lst['t']=1         OPT_VALUE_lst['t']="$TPLG"
+
+OPT_OPT_lst['i']='sleep'    OPT_DESC_lst['i']='interval time for stop/start'
+OPT_PARM_lst['i']=1         OPT_VALUE_lst['i']=0.5
+
+OPT_OPT_lst['d']='duration' OPT_DESC_lst['d']='duration time'
+OPT_PARM_lst['d']=1         OPT_VALUE_lst['d']=10
+
+OPT_OPT_lst['s']='sof-logger'   OPT_DESC_lst['s']="Open sof-logger trace the data will store at $LOG_ROOT"
+OPT_PARM_lst['s']=0             OPT_VALUE_lst['s']=1
+
+func_opt_parse_option $*
+
+tplg=${OPT_VALUE_lst['t']}
+count=${OPT_VALUE_lst['c']}
+interval=${OPT_VALUE_lst['i']}
+duration=${OPT_VALUE_lst['d']}
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+declare -A CMD FILE
+CMD['playback,both']='aplay'
+FILE['playback,both']='/dev/zero'
+CMD['capture,both']='arecord'
+FILE['capture,both']='/dev/null'
+
+func_stop_start_pipeline()
+{
+    count=1
+    while(true)
+    do
+        ps -ef |grep "$pid" |grep -v grep
+        if [ $? -eq 0 ]; then
+            dlogi "Stop/start count: $count"
+            # stop the pipeline
+            kill -SIGSTOP $pid
+            sleep $interval
+            # start the pipeline
+            kill -SIGCONT $pid
+            sleep $interval
+            let count++
+        else
+        break # aplay/arecord is finished, terminate stop/start test
+        fi
+    done
+}
+
+func_test_pipeline_with_type()
+{
+    func_pipeline_export $tplg "type:$1"
+    for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+    do
+        channel=$(func_pipeline_parse_value $idx channel)
+        rate=$(func_pipeline_parse_value $idx rate)
+        fmt=$(func_pipeline_parse_value $idx fmt)
+        dev=$(func_pipeline_parse_value $idx dev)
+        pcm=$(func_pipeline_parse_value $idx pcm)
+        pipeline_type=$(func_pipeline_parse_value $idx "type")
+
+        dlogi "Testing: run stop/start test on PCM:$pcm,$pipeline_type. Interval time: $interval"
+        dlogc "${CMD[$1]}" -D$dev -r $rate -c $channel -f $fmt -d $duration "${FILE[$1]}" -q &
+        "${CMD[$1]}" -D$dev -r $rate -c $channel -f $fmt -d $duration "${FILE[$1]}" -q & pid=$!
+        # do stop/start test
+        func_stop_start_pipeline
+        # check aplay/arecord return value
+        wait $pid
+        if [ $? != 0 ]; then
+            dloge "$pipeline_type on $pcm failed."
+            exit 1
+        fi
+    done
+}
+
+func_test_pipeline_with_type "playback,both"
+func_test_pipeline_with_type "capture,both"
+
+sof-kernel-log-check.sh $KERNEL_LAST_LINE
```

Comment:
@keqiaozhang, you need to load this function `func_lib_setup_kernel_last_line` to get `$KERNEL_LAST_LINE` value

Reply:
Updated.